### DISCUSSION TITLE
Welcome to art-engine Discussions!

### DISCUSSION THREAD

**Discussion by nftchef** [nftchef/art-engine#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions and find answers in https://github.com/nftchef/art-engine/discussions/categories/help-q-a
  * Share ideas.
  * Engage with other community members.

**Comment by Reymlgz** [comment]
Hi, everyone. Does someone know how to add the rarity percentage in the traits of the JSON?

**Reply by nftchef** [reply]
Rarity can only be calculated after all the generation is done, so this would need to be a two-step process. 
The rarity.js util is currently being rewritten and in beta _alpha_ on the https://github.com/nftchef/art-engine/tree/utils/rarity branch.

The algorithm for calculating rarity in that tool is pretty basic right now, but essentially, you would want to take the data from that output, and loop over all the `.json` files to add a rarity property.

An additional caveat, every rarity tool has its own algorithm (rarity.tools, rarity sniffer, etc.) so it's hard to guarantee an exact match, in which case people may get salty if what is in the metadata json does not match rarity.tools.

**Reply by Reymlgz** [reply]
Thx so much for the answer.

**Comment by nftchef** [comment]
Closing this announcement thread to use the other discussion spaces. 🙇
